upgrade/upgrade.go

```diff
@@ -18,25 +18,89 @@ package upgrade
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
+	"cosmossdk.io/core/address"
 	"cosmossdk.io/log"
 	upgradetypes "cosmossdk.io/x/upgrade/types"
+	dollarkeeper "dollar.noble.xyz/keeper"
+	dollartypes "dollar.noble.xyz/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	authoritykeeper "github.com/noble-assets/authority/keeper"
+	swapkeeper "swap.noble.xyz/keeper"
 )
 
 func CreateUpgradeHandler(
 	mm *module.Manager,
 	cfg module.Configurator,
 	logger log.Logger,
+	addressCodec address.Codec,
+	authorityKeeper *authoritykeeper.Keeper,
+	bankKeeper bankkeeper.Keeper,
+	dollarKeeper *dollarkeeper.Keeper,
+	swapKeeper *swapkeeper.Keeper,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx context.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
 		vm, err := mm.RunMigrations(ctx, cfg, vm)
 		if err != nil {
 			return vm, err
 		}
 
+		err = ClaimSwapPoolYield(ctx, logger, addressCodec, authorityKeeper, bankKeeper, dollarKeeper, swapKeeper)
+		if err != nil {
+			return vm, err
+		}
+
 		logger.Info(UpgradeASCII)
 
 		return vm, nil
 	}
 }
+
+// ClaimSwapPoolYield claims the $USDN yield accrued inside the Noble Swap
+// pools and sends it to the authority address.
+func ClaimSwapPoolYield(
```

Comment:
wouldn't be better to call it `ClaimSwapPoolsYield` since iterates all the pools?

Reply:
Done!